### PR TITLE
Bump Go version to 1.22 & controller-gen to 0.14

### PR DESCRIPTION
**Pull request by roothorp**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Comment by josvazg** [comment]
You need to run `make licenses.csv`

**Comment by josvazg** [comment]
It seems you might also need to update `controller-gen` as well.

**Review by s-urbaniak — APPROVED** [review]

**Review by josvazg — APPROVED** [review]

**Comment by roothorp** [comment]
I think the e2e tests are getting a false cache hit (because we haven't actually changed any dependencies) and are still using 1.21 :(

File: Makefile
Comment (igor-karpukhin) [inline Makefile:116]: This is going to always try to install go-licences, isn't it?

File: Makefile
Comment (igor-karpukhin) [inline Makefile:242]: ...here as well

**Review by igor-karpukhin — COMMENTED** [review]
LGTM in general. Just two questions about the `Makefile` targets

File: Makefile
Comment (roothorp) [reply Makefile:116]: Yeah (the same with controller-gen) but it ensures we're installing the version we want, and as far as I can tell `go install` will just exit if it's already installed

Comment (roothorp) [reply Makefile:116]: to expand; this feels safer, and I had issues when doing this with both `controller-gen` and `go-licenses` because I had old versions installed and `which` doesn't care what version there is, just that it exists.

**Review by igor-karpukhin — APPROVED** [review]

**Comment by josvazg** [comment]
This will need rebase now before merging, but hopefully it will get simpler after the few fixes in the CI.

File: go.mod
Comment (s-urbaniak) [reply go.mod:3]: yes, I realized this as well in atlas CLI :-/

**Review by helderjs — CHANGES_REQUESTED** [review]

**Comment by josvazg** [comment]
We need to run `make recompute-licenses` and update. I would do that after rebasing from the merge of the release, rerun all suite and merge.

File: .github/workflows/test-e2e-gov.yml
Comment (josvazg) [reply .github/workflows/test-e2e-gov.yml:16]: This might be my fault. We were fighting some strange issue with PR target events, and I think I touched more places than needed, like this one. We have removed the PR target issue now, on a separate PR I merged before.

I think we can remove this, but I let it to @roothorp for the final decision.

Comment (josvazg) [inline .github/workflows/test-e2e-gov.yml:16]:
```suggestion
```

**Review by s-urbaniak — APPROVED** [review]
great work, LGTM 👍

**Review by helderjs — APPROVED** [review]

**Pull request merged**
